src/common/checksumalgorithms.h

```diff
@@ -0,0 +1,95 @@
+/*
+ * Copyright (C) by Hannah von Reth <hannah.vonreth@owncloud.com>
+ *
+ * This library is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 2.1 of the License, or (at your option) any later version.
+ *
+ * This library is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this library; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA
+ */
+#pragma once
+
+#include <QCryptographicHash>
+#include <QString>
+
+namespace OCC {
+namespace CheckSums {
+    // Sorted by priority
+    enum class Algorithm {
+        SHA3_256 = QCryptographicHash::Sha3_256,
```

Comment:
Does the sorting have any practical effect? You're assigning the values from Qt, so we are using their sorting (except for the last 3 entries at least).

Edit: I see, priorities are handled below. I guess the comment is just stale. Perhaps you can clarify that this is just sorted for the reader here, and the real priorizing is done below.